packages/video_player/video_player/lib/src/web_vtt.dart

```diff
@@ -0,0 +1,173 @@
+  /// Parses a string into a [ClosedCaptionFile], assuming [fileContents] is in
+  /// the WebVTT file format.
+  /// * See: https://en.wikipedia.org/wiki/WebVTT
+  WebVTTCaptionFile(this.fileContents)
+      : _captions = _parseCaptionsFromWebVTTString(fileContents);
+
+  /// The entire body of the VTT file.
+  final String fileContents;
+
+  @override
+  List<Caption> get captions => _captions;
+
+  final List<Caption> _captions;
+}
+
+List<Caption> _parseCaptionsFromWebVTTString(String file) {
+  final List<Caption> captions = <Caption>[];
+  int number = 1;
+  for (List<String> captionLines in _readWebVTTFile(file)) {
+    if (captionLines.length < 2) continue;
+
+    // Caption has header
+    bool hasHeader = captionLines.length > 2;
+    if (hasHeader) {
+      number = int.parse(captionLines[0]);
+    }
+
+    final int captionNumber = number;
+    final _StartAndEnd startAndEnd = _StartAndEnd.fromWebVTTString(
+        hasHeader ? captionLines[1] : captionLines[0]);
+
+    final String text = captionLines.sublist(hasHeader ? 2 : 1).join('\n');
+
+    //TODO: Handle text format
+    final String textWithoutFormat = _parseHtmlString(text);
+
+    final Caption newCaption = Caption(
+      number: captionNumber,
+      start: startAndEnd.start,
+      end: startAndEnd.end,
+      text: textWithoutFormat,
+    );
+
+    if (newCaption.start != null && newCaption.end != null) {
+      captions.add(newCaption);
+      number++;
+    }
+  }
+
+  return captions;
+}
+
+class _StartAndEnd {
+  final Duration start;
+  final Duration end;
+
+  _StartAndEnd(this.start, this.end);
+
+  // Assumes format from an VTT file.
+  // For example:
+  // 00:09.000 --> 00:11.000
+  static _StartAndEnd fromWebVTTString(String line) {
+    final RegExp format =
+        RegExp(_webVTTTimeStamp + _webVTTArrow + _webVTTTimeStamp);
+
+    if (!format.hasMatch(line)) {
+      return _StartAndEnd(null, null);
+    }
+
+    final List<String> times = line.split(_webVTTArrow);
+
+    final Duration start = _parseWebVTTTimestamp(times[0]);
+    final Duration end = _parseWebVTTTimestamp(times[1]);
+
+    return _StartAndEnd(start, end);
+  }
+}
+
+String _parseHtmlString(String htmlString) {
+  var document = parse(htmlString);
+  String parsedString = parse(document.body.text).documentElement.text;
+  return parsedString;
+}
+
+// Parses a time stamp in an VTT file into a Duration.
+// For example:
+//
+// _parseWebVTTimestamp('00:01:08.430')
+// returns
+// Duration(hours: 0, minutes: 1, seconds: 8, milliseconds: 430)
+Duration _parseWebVTTTimestamp(String timestampString) {
+  if (!RegExp(_webVTTTimeStamp).hasMatch(timestampString)) {
+    return null;
+  }
+
+  final List<String> dotSections = timestampString.split('.');
+  final List<String> hoursMinutesSeconds = dotSections[0].split(':');
+
+  int hours = 0;
+  int minutes = 0;
+  int seconds = 0;
+  List<String> styles;
+
+  if (hoursMinutesSeconds.length > 2) {
```

Comment:
Can you make sure that all three of these cases occur in your `_validVTT` test string?

Reply:
New tests created!